interfaces/builtin/networkmanager.go

```diff
@@ -0,0 +1,409 @@
+/etc/machine-id r,
+
+/run/resolvconf/resolv.conf r,
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+
+dbus (send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member={Request,Release}Name
+   peer=(name=org.freedesktop.DBus),
+
+dbus (receive, send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member=GetConnectionUnixProcessID
+   peer=(label=unconfined),
+
+dbus (receive, send)
+   bus=system
+   path=/org/freedesktop/DBus
+   interface=org.freedesktop.DBus
+   member=GetConnectionUnixUser
+   peer=(label=unconfined),
+
+dbus (send)
+  bus=system
+  path=/org/freedesktop/*
+  interface=org.freedesktop.DBus.Properties
+  peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.freedesktop.NetworkManager",
+
+# Allow traffic to/from our path and interface with any method
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/NetworkManager{,/**}
+    interface=org.freedesktop.NetworkManager*,
+
+# Allow traffic to/from org.freedesktop.DBus for networkManager service
+dbus (receive, send)
+    bus=system
+    path=/
+    interface=org.freedesktop.DBus.**,
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/NetworkManager{,/**}
+    interface=org.freedesktop.DBus.**,
+
+# Sleep monitor inside NetworkManager needs this
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/login1
+    interface=org.freedesktop.login1.Manager,
+
+# Allow access to wpa-supplicant for managing WiFi networks
+dbus (receive, send)
+    bus=system
+    path=/fi/w1/wpa_supplicant1{,/**}
+    interface=fi.w1.wpa_supplicant1*,
+dbus (receive, send)
+    bus=system
+    path=/fi/w1/wpa_supplicant1{,/**}
+    interface=org.freedesktop.DBus.**,
+`)
+
+var networkManagerConnectedPlugAppArmor = []byte(`
+# Description: Allow using NetworkManager service. Reserved because this gives
+#  privileged access to the NetworkManager service.
+# Usage: reserved
+
+#include <abstractions/dbus-strict>
+
+# Allow all access to NetworkManager service
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/NetworkManager{,/**}
+    interface=org.freedesktop.NetworkManager*,
+
+dbus (receive)
+    bus=system
+    path=/
+    interface=org.freedesktop.DBus.ObjectManager,
+
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/NetworkManager{,/**}
+    interface=org.freedesktop.DBus.*,
+`)
+
+var networkManagerPermanentSlotSecComp = []byte(`
+# Description: Allow operating as the NetworkManager service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+accept
+accept4
+bind
+connect
+getpeername
+getsockname
+getsockopt
+listen
+recv
+recvfrom
+recvmmsg
+recvmsg
+send
+sendmmsg
+sendmsg
+sendto
+setsockopt
+shutdown
+socketpair
+socket
+# Needed for keyfile settings plugin to allow adding settings
+# for different users.
+chown
```

Comment:
In our case NetworkManager will always chown to root:root but it has support to run chown on the connection settings files in $SNAP_DATA/conf/system-connections to other users but currently only uses it for testing purposes. I would like to go with this for now and limit it to root:root once the seccomp arg filtering branch lands. Is that ok?

Adding those other syscall abbreviations.